### PR TITLE
feat: Add load average to host probe

### PR DESCRIPTION
**Pull request by haoel**

### CODE REVIEW

File: docs/Manual.md [hunk 1]
```diff
@@ -1064,9 +1065,13 @@ host:
         cpu: 0.80  # cpu usage  80%
         mem: 0.70  # memory usage 70%
         disk: 0.90  # disk usage 90%
+        load: # load average - Note: the actual load would be dived by cpu core number, the threshold won't consider the cpu core number.
```
Comment (localvar) [inline docs/Manual.md:1068]:
```suggestion
        load: # load average - Note: the actual load would be divided by cpu core number, the threshold won't consider the cpu core number.
```

File: resources/config.yaml [hunk 2]
```diff
@@ -180,9 +180,13 @@ http: # http probes
 #         cpu: 0.80  # cpu usage  80%
 #         mem: 0.70  # memory usage 70%
 #         disk: 0.90  # disk usage 90%
+#         load: # load average - Note: the actual load would be dived by cpu core number, the threshold won't consider the cpu core number.
```
Comment (localvar) [inline resources/config.yaml:183]:
```suggestion
#         load: # load average - Note: the actual load would be divided by cpu core number, the threshold won't consider the cpu core number.
```

File: docs/Manual.md [hunk 3]
```diff
@@ -1042,8 +1042,9 @@ The host resource usage probe allows for collecting information and alerting whe
 The resources currently monitored include CPU, memory and disk utilization. The probe status is considered as `down` when any value exceeds its defined threshold.
 
 > **Note**:
-> - The host running EaseProbe needs the following commands to be installed on the remote system that will be monitored: `top`, `df`, `free`, `awk`, `grep`, `tr`, and `hostname` (check the [source code](./probe/host/host.go) for more details on this works and/or modify its behavior).
+> - The host running EaseProbe needs the following commands to be installed on the remote system that will be monitored: `top`, `df`, `free`, `awk`, `grep`, `tr`, `cat` and `hostname` (check the [source code](./probe/host/host.go) for more details on this works and/or modify its behavior).
 > - The disk usage check is limited to the root filesystem only with the following command `df -h /`.
+> - The actual load would be dived by cpu core number, the threshold won't consider the cpu core number.
```
Comment (proditis) [inline docs/Manual.md:1047]:
```suggestion
> - The actual load would be divided by cpu core number, the threshold won't consider the cpu core number (requires proc filesystem support).
```

**Review by proditis — APPROVED** [review]
one more dived escaped :smile_cat: 
LGTM

**Review by samanhappy — APPROVED** [review]

File: probe/host/disk.go [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+// CheckThreshold check the cpu usage
+func (d *Disks) CheckThreshold() (bool, string) {
+	lowDisks := []string{}
+	for _, disk := range d.Usage {
+		if d.Threshold > 0 && d.Threshold <= disk.Usage/100 {
+			lowDisks = append(lowDisks, disk.Tag)
+		}
+	}
+	if len(lowDisks) > 0 {
+		return false, fmt.Sprintf("Disk Space Low! - [%s]", strings.Join(lowDisks, ", "))
```
Comment (proditis) [inline probe/host/disk.go:104]:
```suggestion
		return false, fmt.Sprintf("Disk Space threshold alert! - [%s]", strings.Join(lowDisks, ", "))
```

File: probe/host/load.go [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+
+// Config returns the config of the cpu
+func (l *Load) Config(s *Server) error {
+	l.Metrics = make(map[string]float64)
+	if s.Threshold.Load == nil {
+		s.Threshold.Load = make(map[string]float64)
+		s.Threshold.Load["m1"] = DefaultLoadThreshold
+		s.Threshold.Load["m5"] = DefaultLoadThreshold
+		s.Threshold.Load["m15"] = DefaultLoadThreshold
+		log.Debugf("[%s / %s] All of load average threshold is not set, use default value: %.2f", s.ProbeKind, s.ProbeName, DefaultLoadThreshold)
```
Comment (proditis) [inline probe/host/load.go:54]:
```suggestion
		log.Debugf("[%s / %s] All of load average threshold is not set, using default value: %.2f", s.ProbeKind, s.ProbeName, DefaultLoadThreshold)
```

File: probe/host/load.go [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+	}
+	return "Load: " + strings.Join(loadAvg, "/")
+}
+
+// CheckThreshold check the cpu usage
+func (l *Load) CheckThreshold() (bool, string) {
+	for k, v := range l.Metrics {
+		// normalize the load average to 1 cpu core
+		if v/float64(l.Core) > l.Threshold[k] {
+			return false, fmt.Sprintf("Load Average %s High! - %.2f", k, v)
```
Comment (proditis) [inline probe/host/load.go:113]:
```suggestion
			return false, fmt.Sprintf("Load Average threshold %s alert! - %.2f", k, v)
```

File: probe/host/mem.go [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+// OutputLines returns the lines of command output
+func (m *Mem) OutputLines() int {
+	return 1
+}
+
+// Config returns the config of the cpu
+func (m *Mem) Config(s *Server) error {
+	if s.Threshold.Mem == 0 {
+		s.Threshold.Mem = DefaultMemThreshold
+		log.Debugf("[%s / %s] Memory threshold is not set, use default value: %.2f", s.ProbeKind, s.ProbeName, s.Threshold.Mem)
```
Comment (proditis) [inline probe/host/mem.go:48]:
```suggestion
		log.Debugf("[%s / %s] Memory threshold is not set, using default value: %.2f", s.ProbeKind, s.ProbeName, s.Threshold.Mem)
```

File: probe/host/mem.go [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+
+// UsageInfo returns the usage info of the memory
+func (m *Mem) UsageInfo() string {
+	return fmt.Sprintf("Memory: %.2f%%", m.Usage)
+}
+
+// CheckThreshold check the cpu usage
+func (m *Mem) CheckThreshold() (bool, string) {
+	if m.Threshold > 0 && m.Threshold <= m.Usage/100 {
+		return false, "Memory Shortage!"
```
Comment (proditis) [inline probe/host/mem.go:82]:
```suggestion
		return false, "Memory threshold alert!"
```

File: probe/host/load.go [hunk 9]
```diff
@@ -0,0 +1,135 @@
...
+	for k, v := range s.Threshold.Load {
+		s.Threshold.Load[strings.ToLower(k)] = v
+	}
+	if _, ok := s.Threshold.Load["m1"]; !ok {
+		s.Threshold.Load["m1"] = DefaultLoadThreshold
+		log.Debugf("[%s / %s] Load average threshold for m1 is not set, use default value: %.2f", s.ProbeKind, s.ProbeName, s.Threshold.Load["m1"])
+	}
+	if _, ok := s.Threshold.Load["m5"]; !ok {
+		s.Threshold.Load["m5"] = DefaultLoadThreshold
+		log.Debugf("[%s / %s] Load average threshold for m5 is not set, use default value: %.2f", s.ProbeKind, s.ProbeName, s.Threshold.Load["m5"])
```
Comment (proditis) [inline probe/host/load.go:68]:
```suggestion
		log.Debugf("[%s / %s] Load average threshold for m5 is not set, using default value: %.2f", s.ProbeKind, s.ProbeName, s.Threshold.Load["m5"])
```

File: probe/host/load.go [hunk 10]
```diff
@@ -0,0 +1,135 @@
...
+		l.SetThreshold(&s.Threshold)
+		return nil
+	}
+
+	for k, v := range s.Threshold.Load {
+		s.Threshold.Load[strings.ToLower(k)] = v
+	}
+	if _, ok := s.Threshold.Load["m1"]; !ok {
+		s.Threshold.Load["m1"] = DefaultLoadThreshold
+		log.Debugf("[%s / %s] Load average threshold for m1 is not set, use default value: %.2f", s.ProbeKind, s.ProbeName, s.Threshold.Load["m1"])
```
Comment (proditis) [inline probe/host/load.go:64]:
```suggestion
		log.Debugf("[%s / %s] Load average threshold for m1 is not set, using default value: %.2f", s.ProbeKind, s.ProbeName, s.Threshold.Load["m1"])
```

File: probe/host/load.go [hunk 11]
```diff
@@ -0,0 +1,135 @@
...
+	}
+	if _, ok := s.Threshold.Load["m5"]; !ok {
+		s.Threshold.Load["m5"] = DefaultLoadThreshold
+		log.Debugf("[%s / %s] Load average threshold for m5 is not set, use default value: %.2f", s.ProbeKind, s.ProbeName, s.Threshold.Load["m5"])
+	}
+	if _, ok := s.Threshold.Load["m15"]; !ok {
+		s.Threshold.Load["m15"] = DefaultLoadThreshold
+		log.Debugf("[%s / %s] Load average threshold for m15 is not set, use default value: %.2f", s.ProbeKind, s.ProbeName, s.Threshold.Load["m15"])
```
Comment (proditis) [inline probe/host/load.go:72]:
```suggestion
		log.Debugf("[%s / %s] Load average threshold for m15 is not set, using default value: %.2f", s.ProbeKind, s.ProbeName, s.Threshold.Load["m15"])
```

File: probe/host/disk.go [hunk 12]
```diff
@@ -0,0 +1,136 @@
...
+// Config returns the config of the cpu
+func (d *Disks) Config(s *Server) error {
+	if len(s.Disks) == 0 {
+		s.Disks = []string{"/"}
+	}
+	d.Mount = []string{}
+	d.Mount = append(d.Mount, s.Disks...)
+	if s.Threshold.Disk == 0 {
+		s.Threshold.Disk = DefaultDiskThreshold
+		log.Debugf("[%s / %s] Disk threshold is not set, use default value: %.2f", s.ProbeKind, s.ProbeName, s.Threshold.Disk)
```
Comment (proditis) [inline probe/host/disk.go:54]:
```suggestion
		log.Debugf("[%s / %s] Disk threshold is not set, using default value: %.2f", s.ProbeKind, s.ProbeName, s.Threshold.Disk)
```

File: probe/host/cpu.go [hunk 13]
```diff
@@ -0,0 +1,148 @@
...
+// OutputLines returns the lines of command output
+func (c *CPU) OutputLines() int {
+	return 1
+}
+
+// Config returns the config of the cpu
+func (c *CPU) Config(s *Server) error {
+	if s.Threshold.CPU == 0 {
+		s.Threshold.CPU = DefaultCPUThreshold
+		log.Debugf("[%s / %s] CPU threshold is not set, use default value: %.2f", s.ProbeKind, s.ProbeName, s.Threshold.CPU)
```
Comment (proditis) [inline probe/host/cpu.go:57]:
```suggestion
		log.Debugf("[%s / %s] CPU threshold is not set, using default value: %.2f", s.ProbeKind, s.ProbeName, s.Threshold.CPU)
```

File: probe/host/mem.go (hunk 7 above)
Comment (samanhappy) [reply probe/host/mem.go:48]: it looks like this suggestion is missing

**Review by localvar — APPROVED** [review]

**Pull request merged**